File: sets.py
"""
"%ProgramFiles%\Python38\python.exe"
import os
curDir = 'D:\\Source\\Repos\\cryptopals\\'
exec(open(os.path.join(curDir, 'sets.py')).read())
"""

import pathlib
import importlib.util
import os
curDir = 'D:\\Source\\Repos\\cryptopals\\'

exec(open(os.path.join(curDir, 'utility.py')).read())
exec(open(os.path.join(curDir, '1.py')).read())
exec(open(os.path.join(curDir, '2.py')).read())
exec(open(os.path.join(curDir, '3.py')).read())
exec(open(os.path.join(curDir, '4.py')).read())
exec(open(os.path.join(curDir, '5.py')).read())
exec(open(os.path.join(curDir, '6.py')).read())
exec(open(os.path.join(curDir, '7.py')).read())
exec(open(os.path.join(curDir, '8.py')).read())
exec(open(os.path.join(curDir, '9.py')).read())

#def loadImp(name):
#  import imp
#  imp.load_source(name, os.path.join(curDir, name = ".py"))

#def loadMod(name):
#  spec = importlib.util.spec_from_file_location(name, os.path.join(curDir, name + ".py"))
#  mod = importlib.util.module_from_spec(spec)
#  spec.loader.exec_module(mod)
#  return mod

def readChallengeFile(fileName):
  with open(os.path.join(curDir, fileName), "r") as f:
    return f.readlines()
  return []

def runSet(setNum, curSet):
  bPassAll = True
  for s in curSet:
    bRes = s[0]()
    if not bRes: bPassAll = False
    print("%s challenge %s.%s" % ("Passed" if bRes else "Failed", setNum, s[1]))
  if bPassAll: print("All challenges passed in set %s" % setNum)
  
#util, s1, s2 = loadMod("utility"), loadMod("1"), loadMod("2")

set1 = [(challenge1, 1), (challenge2, 2), (challenge3, 3), (challenge4, 4),
        (challenge5, 5), (challenge6, 6), (challenge7, 7), (challenge8, 8)]
        
set2 = [(challenge9, 9),   (challenge10, 10), (challenge11, 11),
        (challenge12, 12), (challenge13, 13), (challenge14, 14),
        (challenge15, 15), (challenge16, 16)]

set3 = [(challenge17, 17), (challenge18, 18), (challenge19, 19),
        (challenge20, 20), (challenge21, 21), (challenge22, 22),
        (challenge23, 23), (challenge24, 24)]

set4 = [(challenge25, 25), (challenge26, 26), (challenge27, 27),
        (challenge28, 28), (challenge29, 29), (challenge30, 30),
        (challenge31, 31), (challenge32, 32)]

set5 = [(challenge33, 33), (challenge34, 34), (challenge35, 35),
        (challenge36, 36), (challenge37, 37), (challenge38, 38),
        (challenge39, 39), (challenge40, 40)]

set6 = [(challenge41, 41), (challenge42, 42), (challenge43, 43),
        (challenge44, 44), (challenge45, 45), (challenge46, 46),
        (challenge47, 47), (challenge48, 48)]

set7 = [(challenge49, 49), (challenge50, 50), (challenge51, 51),
        (challenge52, 52), (challenge53, 53), (challenge54, 54),
        (challenge55, 55), (challenge56, 56)]

set8 = [(challenge57, 57), (challenge58, 58), (challenge59, 59),
        (challenge60, 60), (challenge61, 61), (challenge62, 62),
        (challenge63, 63), (challenge64, 64)]

set9 = [(challenge65, 65), (challenge66, 66)]

testUtility()
#runSet(1, set1)
#runSet(2, set2)
#runSet(3, set3)
#runSet(4, set4)
#runSet(5, set5)
#runSet(6, set6)
#runSet(7, set7)
#runSet(8, set8)
challenge59()
#runSet(9, set9)

File: sets.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using static Cryptopals.Crypto;
using static Cryptopals.Utility;

namespace Cryptopals
{
    class sets
    {
        static public string[] ReadChallengeFile(string fileName)
        {
            return System.IO.File.ReadAllLines("../../" + fileName);
        }
        static public void RunSet(int setNum, Tuple<Func<bool>, int>[] curSet)
        {
            bool bPassAll = true;
            foreach (dynamic s in curSet)
            {
                bool bRes = s.Item1();
                if (!bRes) bPassAll = false;
                Console.WriteLine((bRes ? "Passed" : "Failed") + " challenge " + setNum.ToString() + "." + s.Item2.ToString());
            }
            if (bPassAll) Console.WriteLine("All challenges passed in set " + setNum.ToString());
        }
        static Tuple<Func<bool>, int>[] Set1 = {
            new Tuple<Func<bool>, int>(Challenge1, 1),
            new Tuple<Func<bool>, int>(Challenge2, 2),
            new Tuple<Func<bool>, int>(Challenge3, 3),
            new Tuple<Func<bool>, int>(Challenge4, 4),
            new Tuple<Func<bool>, int>(Challenge5, 5),
            new Tuple<Func<bool>, int>(Challenge6, 6),
            new Tuple<Func<bool>, int>(Challenge7, 7),
            new Tuple<Func<bool>, int>(Challenge8, 8)};
        static Tuple<Func<bool>, int>[] Set2 = {
            new Tuple<Func<bool>, int>(Challenge9, 9),
            new Tuple<Func<bool>, int>(Challenge10, 10),
            new Tuple<Func<bool>, int>(Challenge11, 11),
            new Tuple<Func<bool>, int>(Challenge12, 12),
            new Tuple<Func<bool>, int>(Challenge13, 13),
            new Tuple<Func<bool>, int>(Challenge14, 14),
            new Tuple<Func<bool>, int>(Challenge15, 15),
            new Tuple<Func<bool>, int>(Challenge16, 16)
        };
        static Tuple<Func<bool>, int>[] Set3 = {
            new Tuple<Func<bool>, int>(Challenge17, 17),
            new Tuple<Func<bool>, int>(Challenge18, 18),
            new Tuple<Func<bool>, int>(Challenge19, 19),
            new Tuple<Func<bool>, int>(Challenge20, 20),
            new Tuple<Func<bool>, int>(Challenge21, 21),
            new Tuple<Func<bool>, int>(Challenge22, 22),
            new Tuple<Func<bool>, int>(Challenge23, 23),
            new Tuple<Func<bool>, int>(Challenge24, 24)
        };
        static Tuple<Func<bool>, int>[] Set4 = {
            new Tuple<Func<bool>, int>(Challenge25, 25),
            new Tuple<Func<bool>, int>(Challenge26, 26),
            new Tuple<Func<bool>, int>(Challenge27, 27),
            new Tuple<Func<bool>, int>(Challenge28, 28),
            new Tuple<Func<bool>, int>(Challenge29, 29),
            new Tuple<Func<bool>, int>(Challenge30, 30),
            new Tuple<Func<bool>, int>(Challenge31, 31),
            new Tuple<Func<bool>, int>(Challenge32, 32)
        };
        static Tuple<Func<bool>, int>[] Set5 = {
            new Tuple<Func<bool>, int>(Challenge33, 33),
            new Tuple<Func<bool>, int>(Challenge34, 34),
            new Tuple<Func<bool>, int>(Challenge35, 35),
            new Tuple<Func<bool>, int>(Challenge36, 36),
            new Tuple<Func<bool>, int>(Challenge37, 37),
            new Tuple<Func<bool>, int>(Challenge38, 38),
            new Tuple<Func<bool>, int>(Challenge39, 39),
            new Tuple<Func<bool>, int>(Challenge40, 40)
        };
        static Tuple<Func<bool>, int>[] Set6 = {
            new Tuple<Func<bool>, int>(Challenge41, 41),
            new Tuple<Func<bool>, int>(Challenge42, 42),
            new Tuple<Func<bool>, int>(Challenge43, 43),
            new Tuple<Func<bool>, int>(Challenge44, 44),
            new Tuple<Func<bool>, int>(Challenge45, 45),
            new Tuple<Func<bool>, int>(Challenge46, 46),
            new Tuple<Func<bool>, int>(Challenge47, 47),
            new Tuple<Func<bool>, int>(Challenge48, 48)
        };
        static Tuple<Func<bool>, int>[] Set7 = {
            new Tuple<Func<bool>, int>(Challenge49, 49),
            new Tuple<Func<bool>, int>(Challenge50, 50),
            new Tuple<Func<bool>, int>(Challenge51, 51),
            new Tuple<Func<bool>, int>(Challenge52, 52),
            new Tuple<Func<bool>, int>(Challenge53, 53),
            new Tuple<Func<bool>, int>(Challenge54, 54),
            new Tuple<Func<bool>, int>(Challenge55, 55),
            new Tuple<Func<bool>, int>(Challenge56, 56)
        };
        static Tuple<Func<bool>, int>[] Set8 = {
            new Tuple<Func<bool>, int>(Challenge57, 57),
            new Tuple<Func<bool>, int>(Challenge58, 58),
            new Tuple<Func<bool>, int>(Challenge59, 59),
            new Tuple<Func<bool>, int>(Challenge60, 60),
            new Tuple<Func<bool>, int>(Challenge61, 61),
            new Tuple<Func<bool>, int>(Challenge62, 62),
            new Tuple<Func<bool>, int>(Challenge63, 63),
            new Tuple<Func<bool>, int>(Challenge64, 64)
        };
        static Tuple<Func<bool>, int>[] Set9 = {
            new Tuple<Func<bool>, int>(Challenge65, 65),
            new Tuple<Func<bool>, int>(Challenge66, 66),
        };
        static void Main(string[] args)
        {
            //testMul();
            //RunSet(1, Set1);
            //RunSet(2, Set2);
            //RunSet(3, Set3);
            //RunSet(4, Set4);
            //RunSet(5, Set5);
            //RunSet(6, Set6);
            //RunSet(7, Set7);
            //RunSet(8, Set8);
            RunSet(9, Set9);
            Console.ReadKey();
        }
    }
}
